context/app/static/js/components/Detail/Dataset/Dataset.jsx

```diff
@@ -25,19 +25,24 @@ function AssaySpecificItem(props) {
 }
 
 function SummaryData(props) {
-  const { data_types, origin_sample } = props;
-  // TODO: ES should consistenty return array. Waiting for reindex.
-  const data_types_array = data_types.constructor.name === 'Array' ? data_types : [data_types];
+  const { data_types, mapped_data_types, origin_sample } = props;
+
+  // TODO: This was an array, but in production it's a single value.
+  // Simplify when it's single valued in all environments.
+
+  // NOTE: This is the one place we use unmapped values,
+  // because they are better than human-readable strings as URL fragments.
+
+  const typeCodes = Array.isArray(data_types) ? data_types : [data_types];
+  const typeNames = Array.isArray(mapped_data_types) ? mapped_data_types : [mapped_data_types];
+  const codesNames = typeCodes.map((code, i) => {
+    return { code, name: typeNames[i] };
+  });
   return (
     <>
-      {data_types && data_types.length > 0 && (
-        <AssaySpecificItem>
-          {data_types_array.map((data_type, i) => [
-            i > 0 && ' / ',
-            <a href={`/docs/assays#${data_type}`}>{data_type}</a>,
-          ])}
-        </AssaySpecificItem>
-      )}
+      <AssaySpecificItem>
+        {codesNames.map(({ code, name }, i) => [i > 0 && ' / ', <a href={`/docs/assays#${code}`}>{name}</a>])}
```

Comment:
We probably want to make the color of this link consistent with other links.

Reply:
Filed #665